lib/storage/providers/SQLiteStorage.js

```diff
@@ -0,0 +1,118 @@
+    getItem(key) {
+        return db.executeAsync('SELECT record_key, valueJSON FROM magic_map WHERE record_key=?;', [key]).then(({rows}) => {
+            if (rows.length === 0) {
+                return null;
+            }
+            const result = rows.item(0);
+            return JSON.parse(result.valueJSON);
+        });
+    },
+
+    /**
+    * Get multiple key-value pairs for the given array of keys in a batch
+    * @param {String[]} keys
+    * @return {Promise<Array<[key, value]>>}
+    */
+    multiGet(keys) {
+        const placeholders = _.map(keys, () => '?').join(',');
+        const command = `SELECT record_key, valueJSON FROM magic_map WHERE record_key IN (${placeholders});`;
+        return db.executeAsync(command, keys)
+            .then(({rows}) => {
+                // eslint-disable-next-line no-underscore-dangle
+                const result = _.map(rows._array, row => [row.record_key, JSON.parse(row.valueJSON)]);
+                return result;
+            });
+    },
+
+    /**
+    * Sets the value for a given key. The only requirement is that the value should be serializable to JSON string
+    * @param {String} key
+    * @param {*} value
+    * @return {Promise<void>}
+    */
+    setItem(key, value) {
+        return db.executeAsync('REPLACE INTO magic_map (record_key, valueJSON) VALUES (?, ?);', [key, JSON.stringify(value)]);
+    },
+
+    /**
+    * Stores multiple key-value pairs in a batch
+    * @param {Array<[key, value]>} pairs
+    * @return {Promise<void>}
+    */
+    multiSet(pairs) {
+        const stringifiedPairs = _.map(pairs, pair => [
+            pair[0],
+            JSON.stringify(pair[1]),
+        ]);
+        return db.executeBatchAsync([['REPLACE INTO magic_map (record_key, valueJSON) VALUES (?, json(?));', stringifiedPairs]]);
+    },
+
+    /**
+    * Multiple merging of existing and new values in a batch
+    * @param {Array<[key, value]>} pairs
+    * @return {Promise<void>}
+    */
+    multiMerge(pairs) {
+        // Note: We use `ON CONFLICT DO UPDATE` here instead of `INSERT OR REPLACE INTO`
+        // so the new JSON value is merged into the old one if there's an existing value
+        const query = `INSERT INTO magic_map (record_key, valueJSON) 
+             VALUES (:key, JSON(:value)) 
+             ON CONFLICT DO UPDATE 
+             SET valueJSON = JSON_PATCH(valueJSON, JSON(:value)); 
+        `;
+        const queryArguments = _.map(pairs, (pair) => {
+            const value = JSON.stringify(pair[1]);
+            return [pair[0], value];
+        });
+        return db.executeBatchAsync([[query, queryArguments]]);
+    },
+
+    /**
+    * Returns all keys available in storage
+    * @returns {Promise<String[]>}
+    */
+    getAllKeys: () => db.executeAsync('SELECT record_key FROM magic_map;').then(({rows}) => {
+        // eslint-disable-next-line no-underscore-dangle
+        const result = _.map(rows._array, row => row.record_key);
+        return result;
```

Comment:
```suggestion
        return result = _.map(rows._array, row => row.record_key);
```

Reply:
I wanted to do that but then eslint forces me to remove return keyword to what would require me to move (// eslint-disable-next-line no-underscore-dangle) between definition of the function and JSdoc.